utils.go

```diff
@@ -873,3 +873,19 @@ func copySortedFunc[E any](slice []E, less func(a, b E) bool) []E {
 	slices.SortFunc(sliceCopy, less)
 	return sliceCopy
 }
+
+// Get the nodes needed to prove a target leaf.
+func (p *Pollard) getNodesForProof(position uint64) ([]Leaf, []uint64, error) {
```

Comment:
There's no need for this function. As mentioned above, you can just use the `Verify` method on pollard instead.